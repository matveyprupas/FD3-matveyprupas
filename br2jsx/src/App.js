import React from "react";
import BR2JSX from "./br2jsx";

import './App.css';

class App extends React.Component {
  render() {
    let text="первый<br>второй<br/>третий<br />последний";
    return <BR2JSX text={text}/>;
  }
}

export default App;
