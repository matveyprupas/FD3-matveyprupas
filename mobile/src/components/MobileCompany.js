import React from 'react';
import PropTypes from 'prop-types';
import MobileClient from './MobileClient.js';
import EditMode from './editMode';
import {mobileEvents} from './events';

import './MobileCompany.css';

class MobileCompany extends React.PureComponent {

  static propTypes = {
    name: PropTypes.string.isRequired,
    clients:PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        lastname: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        secondname: PropTypes.string.isRequired,
        balance: PropTypes.number.isRequired,
      })
    ),
  };

  state = {
    name: this.props.name,
    allClients: this.props.clients,
    showedClients: this.props.clients,
    clientsID: this.props.clients.map( el => el.id),
    editMode: false,
    editedClient: null,
  };

  setName = (str) => {
    this.setState( {name: str} );
  };

  setName1 = () => {
    this.setName("MTC");
  };

  setName2 = () => {
    this.setName("Velcom");
  };

  showAllClients = () => {
    this.setState( {showedClients: this.state.allClients} );
  };

  showActiveClients = () => {
    let sortedClients = [...this.state.allClients].filter( el => el.balance > 0 );
    console.log(sortedClients);
    this.setState( {showedClients: sortedClients} );
  };

  showDeactiveClients = () => {
    let sortedClients = [...this.state.allClients].filter( el => el.balance <= 0 );
    console.log(sortedClients);
    this.setState( {showedClients: sortedClients} );
  };

  activateEditMode = (obj) => {
    this.setState( {editMode: true, editedClient: obj} );
  };

  deactivateEditMode = () => {
    this.setState( {editMode: false} );
  };

  addNewClient = (obj) => {
    if (!this.state.clientsID.includes(obj.id)) {
      this.setState( {showedClients: [...this.state.showedClients, obj], clientsID: [...this.state.clientsID, obj.id]} );
    }
  };

  componentDidMount = () => {
    mobileEvents.addListener('editedClientInfo',this.addNewClient);
  };

  componentWillUnmount = () => {
    mobileEvents.removeListener('editedClientInfo',this.addNewClient);
  };

  render() {

    console.log("MobileCompany render");

    var clientsCode=this.state.showedClients.map( c =>
      <MobileClient key={c.id} clientInfo={c} cbActivateEditMode = {this.activateEditMode} />
    );

    return (
      <div className='mobile-company'>
        <input type="button" value="=МТС" onClick={this.setName1} />
        <input type="button" value="=Velcom" onClick={this.setName2} />

        <h1 className='mobile-company__name'>Компания &laquo;{this.state.name}&raquo;</h1>
        <hr />

        <input type="button" value="All" onClick={this.showAllClients} />
        <input type="button" value="Active" onClick={this.showActiveClients} />
        <input type="button" value="Deactive" onClick={this.showDeactiveClients} />
        <hr />

        <div className='mobile-company__client'>
          <div className="mobile-client__row">
            <div className='mobile-client__lastname'>{"Фамилия"}</div>
            <div className='mobile-client__name'>{"Имя"}</div>
            <div className='mobile-client__fathername'>{"Отчество"}</div>
            <div className='mobile-client__balance'>{"Баланс"}</div>
            <div className='mobile-client__status'>{"Статус"}</div>
            <div className='mobile-client__edit'>{"Редактировать"}</div>
          </div>

          {clientsCode}
        </div>
        <input type="button" value="Add client" onClick= {()=>this.activateEditMode({newClient: true})}/>
        {
          this.state.editMode &&
          <EditMode  key={this.state.editedClient.id} id={this.state.editedClient.id} lastname={this.state.editedClient.lastname} name={this.state.editedClient.name} secondname={this.state.editedClient.secondname} balance={this.state.editedClient.balance} newClient={this.state.editedClient.newClient} clientsID={this.state.clientsID} cbDeactivateEditMode = {this.deactivateEditMode} />
        }
      </div>
    );

  }

}

export default MobileCompany;
