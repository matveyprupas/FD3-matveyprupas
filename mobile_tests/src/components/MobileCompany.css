.mobile-company__client .mobile-client__row:first-of-type {
  font-weight: 700;
}

.mobile-client__row {
  /* max-width: 800px; */
  display: flex;
  justify-content: space-between;
}

.mobile-client__row:nth-of-type(2n + 1) {
  background-color: lightgrey;
}

.mobile-client__row > * {
  width: 200px;
  padding: 1rem 0;
}
