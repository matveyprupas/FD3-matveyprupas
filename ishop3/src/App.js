import React from 'react';
import PropTypes from 'prop-types';
import './App.css';
import Goods from './components/goods';
import GoodCard from './components/goodCard';
import GoodEdit from './components/goodEdit';


class Shop3 extends React.Component {

  static propTypes = {
    shopName: PropTypes.string.isRequired,
    goodsArr: PropTypes.array.isRequired,
  }

  state = {
    goodsArr: this.props.goodsArr,
    choosedCode: 0,
    editMode: false,
    idArr: [],
  }

  chooseGood = (obj) => {
    if (!this.state.editMode) this.setState( {choosedCode: obj.code, editMode: false} );
  }

  removeGood = (obj) => {
    this.setState({goodsArr: this.state.goodsArr.filter(el => obj.code !== el.code)});
  }

  activateEditMode = (obj) => {
    this.setState({choosedCode: obj.code, editMode: true});
  }

  activateEditModeWithoutObj = () => {
    let code;
    for (let i = 1; i < this.state.idArr.length+2; i++ ) {
      if(!this.state.idArr.includes(i)) {
        code = i;
      }
    }
    this.setState({choosedCode: code, editMode: true});
  }

  deactivateEditMode = () => {
    this.setState({choosedCode: 0, editMode: false});
  }

  componentDidUpdate = (prevProps, prevState) => {
    if (this.state.goodsArr.length !== prevState.goodsArr.length) this.setState( {idArr: this.state.goodsArr.map(el => el.code)} );
  }

  componentDidMount = () => {
    this.setState( {idArr: this.state.goodsArr.map(el => el.code)} );
  }

  saveNewProductValues = (obj) => {
    if (this.state.idArr.includes(obj.code)) {
      this.updateOldGood(obj);
    } else {
      this.addNewGood(obj);
    }
  }

  addNewGood = (obj) => {
    this.setState ( {goodsArr: [...this.state.goodsArr, obj], editMode: false} );
  }

  updateOldGood = (obj) => {
    let index = this.state.goodsArr.findIndex((el, i) => obj.code === el.code);

    let updatedGoodsArr = [...this.state.goodsArr];
    updatedGoodsArr[index] = obj;

    this.setState ( {goodsArr: updatedGoodsArr, editMode: false} );
  }

  render() {
    let goodsArrJSX = this.state.goodsArr.map(el => {
      return (
        <Goods key = {el.code} code = {el.code} name = {el.name} cost = {el.cost} imageLink = {el.imageLink} left = {el.left} cbChooseGood = {this.chooseGood} cbRemoveGood = {this.removeGood} cbActivateEditMode = {this.activateEditMode} choose = {this.state.choosedCode === el.code ? true : false} editMode = {this.state.editMode} />
      ); 
    });

    let goodCardProduct = this.state.goodsArr.filter ( el => el.code === this.state.choosedCode )[0];
    
    let goodEditProduct = {...goodCardProduct};

    if(!Object.keys(goodEditProduct).length) goodEditProduct = {
      code: this.state.choosedGood,
      name: undefined, 
      cost: undefined, 
      imageLink: undefined, 
      left: undefined,
    };

    return (
      <div className="goods__frame">
        <h1 className="goods__title">{this.props.shopName}</h1>
        <div className="goods__wrap">{goodsArrJSX}</div>
        {
          !this.state.editMode &&
          <input className = "good_del" type = "button" value = "Add product" onClick = {this.activateEditModeWithoutObj} />
        }
        {
          goodCardProduct &&
          !this.state.editMode &&
          <GoodCard code = {goodCardProduct.code} name = {goodCardProduct.name} cost = {goodCardProduct.cost} imageLink = {goodCardProduct.imageLink} left = {goodCardProduct.left} editMode = {this.state.editMode} />
        }
        {
          goodEditProduct &&
          this.state.editMode &&
          <GoodEdit code = {goodEditProduct.code || this.state.choosedCode} name = {goodEditProduct.name} cost = {goodEditProduct.cost} imageLink = {goodEditProduct.imageLink} left = {goodEditProduct.left} idArr = {this.state.idArr} cbDeactivateEditMode = {this.deactivateEditMode} cbSaveNewProductValues = {this.saveNewProductValues} />
        }
      </div>
    )
  }
}

export default Shop3;
