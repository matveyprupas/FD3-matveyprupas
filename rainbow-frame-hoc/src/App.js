import React from "react";
import Fragment from "./fragment";
import DoubleButton from "./DoubleButton";
import {withRainbowFrame} from './withRainbowFrame';

import './App.css';

class App extends React.Component {


  render() {
    let colors=['red','orange', 'yellow','green', '#00BFFF', 'blue', 'purple'];

    // let FramedFragment=withRainbowFrame(colors)(Fragment);
    let FramedFragment=withRainbowFrame(colors)(Fragment);
    let FramedDoubleButton=withRainbowFrame(colors)(DoubleButton);

    return (
      <div>
        <DoubleButton caption1="однажды" caption2="пору" cbPressed={ num => alert(num) } >
          в студёную зимнюю
        </DoubleButton>
        <br />
        <FramedDoubleButton caption1="я из лесу" caption2="мороз" cbPressed={ num => alert(num) }>вышел, был сильный</FramedDoubleButton>
        <br />
        <FramedFragment>
          Hello!
        </FramedFragment>
      </div>
    );
  }
}

export default App;