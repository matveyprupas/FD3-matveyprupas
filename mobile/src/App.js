import React from 'react';
import './App.css';
import MobileCompany from './components/MobileCompany';

function App() {

  let companyName='Velcom';
  let clientsArr=[ 
    {id:101, lastname:"Иванов", name: "Иван", secondname: "Иванович", balance:200}, 
    {id:105, lastname:"Сидоров", name: "Сидор", secondname: "Сидорович", balance:2700}, 
    {id:110, lastname:"Петров", name: "Петр", secondname: "Петрович", balance:-2200},
    {id:120, lastname:"Прупас", name: "Матвей", secondname: "Викторович", balance:100500},
  ];

  return (
    <MobileCompany name={companyName} clients={clientsArr} />
  );
}

export default App;
