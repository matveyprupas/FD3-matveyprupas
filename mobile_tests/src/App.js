import './App.css';
import MobileCompany from './components/MobileCompany';
import clientsHash from './clientsHash.json';

function App() {
  return (
    <MobileCompany name={clientsHash.companyName} clients={clientsHash.clientsArr} />
  );
}

export default App;
