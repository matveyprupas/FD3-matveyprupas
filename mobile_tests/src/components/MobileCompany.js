import React from 'react';
import PropTypes from 'prop-types';
import MobileClient from './MobileClient.js';
import EditMode from './editMode';
import {mobileEvents} from './events';

import './MobileCompany.css';

class MobileCompany extends React.PureComponent {

  static propTypes = {
    name: PropTypes.string.isRequired,
    clients:PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        lastname: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        secondname: PropTypes.string.isRequired,
        balance: PropTypes.number.isRequired,
      })
    ),
  };

  state = {
    name: this.props.name,
    allClients: this.props.clients,
    clientsID: this.props.clients.map( el => el.id),
    editMode: false,
    showMode: "all",
    editedClient: null,
  };

  showAllClients = () => {
    this.setState( {showMode: "all"} );
  };

  showActiveClients = () => {
    this.setState( {showMode: "active"} );
  };

  showDeactiveClients = () => {
    this.setState( {showMode: "deactive"} );
  };

  activateEditMode = (obj) => {
    this.setState( {editMode: true, editedClient: obj} );
  };

  deactivateEditMode = () => {
    this.setState( {editMode: false} );
  };

  addNewClient = (obj) => {
    if (!this.state.clientsID.includes(obj.id)) {
      this.setState( {allClients: [...this.state.allClients, obj], clientsID: [...this.state.clientsID, obj.id]} );
    }
  };

  removeClient = (id) => {
    console.log(this.state);
    if (this.state.clientsID.includes(id)) {
      let sortedClients = this.state.allClients.filter( el => el.id !== id);
      let sortedClientsID = this.state.clientsID.filter( el => el !== id);
      this.setState( {allClients: sortedClients, clientsID: sortedClientsID}, ()=> console.log(this.state) );
    }
    this.deactivateEditMode();
  };

  componentDidMount = () => {
    mobileEvents.addListener('editedClientInfo',this.addNewClient);
  };

  componentWillUnmount = () => {
    mobileEvents.removeListener('editedClientInfo',this.addNewClient);
  };

  render() {

    console.log("MobileCompany render");

    var clientsCode = [];

    if (this.state.showMode === "all") {
      clientsCode=this.state.allClients.map( c =>
        <MobileClient key={c.id} clientInfo={c} cbActivateEditMode = {this.activateEditMode} />
      );
    } else if (this.state.showMode === "active") {
      clientsCode=this.state.allClients.filter(el => el.balance > 0).map( c =>
        <MobileClient key={c.id} clientInfo={c} cbActivateEditMode = {this.activateEditMode} />
      );
    } else if (this.state.showMode === "deactive") {
      clientsCode=this.state.allClients.filter(el => el.balance <= 0).map( c =>
        <MobileClient key={c.id} clientInfo={c} cbActivateEditMode = {this.activateEditMode} />
      );
    }


    return (
      <div className='mobile-company'>
        <h1 className='mobile-company__name'>Компания &laquo;{this.state.name}&raquo;</h1>
        <hr />

        <input type="button" value="All" className="all-btn" onClick={this.showAllClients} />
        <input type="button" value="Active" onClick={this.showActiveClients} />
        <input type="button" value="Deactive" onClick={this.showDeactiveClients} />
        <hr />

        <div className='mobile-company__client'>
          <div className="mobile-client__row">
            <div className='mobile-client__lastname'>{"Фамилия"}</div>
            <div className='mobile-client__name'>{"Имя"}</div>
            <div className='mobile-client__fathername'>{"Отчество"}</div>
            <div className='mobile-client__balance'>{"Баланс"}</div>
            <div className='mobile-client__status'>{"Статус"}</div>
            <div className='mobile-client__edit'>{"Редактировать"}</div>
          </div>

          {clientsCode}
        </div>
        <input type="button" value="Add client" onClick= {()=>this.activateEditMode({newClient: true})}/>
        {
          this.state.editMode &&
          <EditMode  key={this.state.editedClient.id} id={this.state.editedClient.id} lastname={this.state.editedClient.lastname} name={this.state.editedClient.name} secondname={this.state.editedClient.secondname} balance={this.state.editedClient.balance} newClient={this.state.editedClient.newClient} clientsID={this.state.clientsID} cbDeactivateEditMode = {this.deactivateEditMode} cbRemoveClient = {this.removeClient} />
        }
      </div>
    );

  }

}

export default MobileCompany;
