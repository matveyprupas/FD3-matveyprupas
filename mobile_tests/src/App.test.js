import { render, screen } from '@testing-library/react';
import App from './App';
import MobileCompany from './components/MobileCompany';
import EditMode from './components/editMode';
// import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer';
import clientsHash from './clientsHash.json';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

test('работа MobileCompany', () => {
  const component = renderer.create(
    <MobileCompany name={clientsHash.companyName} clients={clientsHash.clientsArr} />
  );

  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  const allBtn = component.root.find( el => el.props.value === 'All' ); 
  const activeBtn = component.root.find( el => el.props.value === 'Active' ); 
  const deactiveBtn = component.root.find( el => el.props.value === 'Deactive' ); 

  activeBtn.props.onClick();
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  deactiveBtn.props.onClick();
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  allBtn.props.onClick();
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('работа EditMode', () => {

  const component = renderer.create(
    <EditMode name={clientsHash.clientsArr[0].name} id={clientsHash.clientsArr[0].id} lastname={clientsHash.clientsArr[0].lastname} secondname={clientsHash.clientsArr[0].secondname} balance={clientsHash.clientsArr[0].balance} />
  );

  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  const allBtn = component.root.find( el => el.props.value === 'Save' ); 
    // console.log("allBtn: ", allBtn);

});



// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
