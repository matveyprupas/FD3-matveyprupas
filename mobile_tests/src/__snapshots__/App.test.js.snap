// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа EditMode 1`] = `
<div
  className="edit-form"
>
  <h2>
    Edit client info
  </h2>
  <div
    className="edit-form__row"
  >
    <label
      htmlFor="edit-form__lastname"
    >
      Lastname
    </label>
    <input
      defaultValue="Иванов"
      id="edit-form__lastname"
      type="text"
    />
  </div>
  <div
    className="edit-form__row"
  >
    <label
      htmlFor="edit-form__name"
    >
      Name
    </label>
    <input
      defaultValue="Иван"
      id="edit-form__name"
      type="text"
    />
  </div>
  <div
    className="edit-form__row"
  >
    <label
      htmlFor="edit-form__secondname"
    >
      Secondname
    </label>
    <input
      defaultValue="Иванович"
      id="edit-form__secondname"
      type="text"
    />
  </div>
  <div
    className="edit-form__row"
  >
    <label
      htmlFor="edit-form__balance"
    >
      Balance
    </label>
    <input
      defaultValue={200}
      id="edit-form__balance"
      type="text"
    />
  </div>
  <input
    id="save-btn"
    onClick={[Function]}
    type="button"
    value="Save"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Del"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Cancel"
  />
</div>
`;

exports[`работа MobileCompany 1`] = `
<div
  className="mobile-company"
>
  <h1
    className="mobile-company__name"
  >
    Компания «
    Velcom
    »
  </h1>
  <hr />
  <input
    onClick={[Function]}
    type="button"
    value="All"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Active"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Deactive"
  />
  <hr />
  <div
    className="mobile-company__client"
  >
    <div
      className="mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Фамилия
      </div>
      <div
        className="mobile-client__name"
      >
        Имя
      </div>
      <div
        className="mobile-client__fathername"
      >
        Отчество
      </div>
      <div
        className="mobile-client__balance"
      >
        Баланс
      </div>
      <div
        className="mobile-client__status"
      >
        Статус
      </div>
      <div
        className="mobile-client__edit"
      >
        Редактировать
      </div>
    </div>
    <div
      className="mobile-client mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Иванов
      </div>
      <div
        className="mobile-client__name"
      >
        Иван
      </div>
      <div
        className="mobile-client__secondname"
      >
        Иванович
      </div>
      <div
        className="mobile-client__balance"
      >
        200
      </div>
      <div
        className="mobile-client__active"
        style={
          Object {
            "backgroundColor": "green",
          }
        }
      >
        active
      </div>
      <input
        onClick={[Function]}
        type="button"
        value="Edit"
      />
    </div>
    <div
      className="mobile-client mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Сидоров
      </div>
      <div
        className="mobile-client__name"
      >
        Сидор
      </div>
      <div
        className="mobile-client__secondname"
      >
        Сидорович
      </div>
      <div
        className="mobile-client__balance"
      >
        2700
      </div>
      <div
        className="mobile-client__active"
        style={
          Object {
            "backgroundColor": "green",
          }
        }
      >
        active
      </div>
      <input
        onClick={[Function]}
        type="button"
        value="Edit"
      />
    </div>
    <div
      className="mobile-client mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Петров
      </div>
      <div
        className="mobile-client__name"
      >
        Петр
      </div>
      <div
        className="mobile-client__secondname"
      >
        Петрович
      </div>
      <div
        className="mobile-client__balance"
      >
        -2200
      </div>
      <div
        className="mobile-client__active"
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      >
        blocked
      </div>
      <input
        onClick={[Function]}
        type="button"
        value="Edit"
      />
    </div>
    <div
      className="mobile-client mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Прупас
      </div>
      <div
        className="mobile-client__name"
      >
        Матвей
      </div>
      <div
        className="mobile-client__secondname"
      >
        Викторович
      </div>
      <div
        className="mobile-client__balance"
      >
        100500
      </div>
      <div
        className="mobile-client__active"
        style={
          Object {
            "backgroundColor": "green",
          }
        }
      >
        active
      </div>
      <input
        onClick={[Function]}
        type="button"
        value="Edit"
      />
    </div>
    <div
      className="mobile-client mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Довгель
      </div>
      <div
        className="mobile-client__name"
      >
        Анастасия
      </div>
      <div
        className="mobile-client__secondname"
      >
        Генриховна
      </div>
      <div
        className="mobile-client__balance"
      >
        0
      </div>
      <div
        className="mobile-client__active"
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      >
        blocked
      </div>
      <input
        onClick={[Function]}
        type="button"
        value="Edit"
      />
    </div>
  </div>
  <input
    onClick={[Function]}
    type="button"
    value="Add client"
  />
</div>
`;

exports[`работа MobileCompany 2`] = `
<div
  className="mobile-company"
>
  <h1
    className="mobile-company__name"
  >
    Компания «
    Velcom
    »
  </h1>
  <hr />
  <input
    onClick={[Function]}
    type="button"
    value="All"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Active"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Deactive"
  />
  <hr />
  <div
    className="mobile-company__client"
  >
    <div
      className="mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Фамилия
      </div>
      <div
        className="mobile-client__name"
      >
        Имя
      </div>
      <div
        className="mobile-client__fathername"
      >
        Отчество
      </div>
      <div
        className="mobile-client__balance"
      >
        Баланс
      </div>
      <div
        className="mobile-client__status"
      >
        Статус
      </div>
      <div
        className="mobile-client__edit"
      >
        Редактировать
      </div>
    </div>
    <div
      className="mobile-client mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Иванов
      </div>
      <div
        className="mobile-client__name"
      >
        Иван
      </div>
      <div
        className="mobile-client__secondname"
      >
        Иванович
      </div>
      <div
        className="mobile-client__balance"
      >
        200
      </div>
      <div
        className="mobile-client__active"
        style={
          Object {
            "backgroundColor": "green",
          }
        }
      >
        active
      </div>
      <input
        onClick={[Function]}
        type="button"
        value="Edit"
      />
    </div>
    <div
      className="mobile-client mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Сидоров
      </div>
      <div
        className="mobile-client__name"
      >
        Сидор
      </div>
      <div
        className="mobile-client__secondname"
      >
        Сидорович
      </div>
      <div
        className="mobile-client__balance"
      >
        2700
      </div>
      <div
        className="mobile-client__active"
        style={
          Object {
            "backgroundColor": "green",
          }
        }
      >
        active
      </div>
      <input
        onClick={[Function]}
        type="button"
        value="Edit"
      />
    </div>
    <div
      className="mobile-client mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Прупас
      </div>
      <div
        className="mobile-client__name"
      >
        Матвей
      </div>
      <div
        className="mobile-client__secondname"
      >
        Викторович
      </div>
      <div
        className="mobile-client__balance"
      >
        100500
      </div>
      <div
        className="mobile-client__active"
        style={
          Object {
            "backgroundColor": "green",
          }
        }
      >
        active
      </div>
      <input
        onClick={[Function]}
        type="button"
        value="Edit"
      />
    </div>
  </div>
  <input
    onClick={[Function]}
    type="button"
    value="Add client"
  />
</div>
`;

exports[`работа MobileCompany 3`] = `
<div
  className="mobile-company"
>
  <h1
    className="mobile-company__name"
  >
    Компания «
    Velcom
    »
  </h1>
  <hr />
  <input
    onClick={[Function]}
    type="button"
    value="All"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Active"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Deactive"
  />
  <hr />
  <div
    className="mobile-company__client"
  >
    <div
      className="mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Фамилия
      </div>
      <div
        className="mobile-client__name"
      >
        Имя
      </div>
      <div
        className="mobile-client__fathername"
      >
        Отчество
      </div>
      <div
        className="mobile-client__balance"
      >
        Баланс
      </div>
      <div
        className="mobile-client__status"
      >
        Статус
      </div>
      <div
        className="mobile-client__edit"
      >
        Редактировать
      </div>
    </div>
    <div
      className="mobile-client mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Петров
      </div>
      <div
        className="mobile-client__name"
      >
        Петр
      </div>
      <div
        className="mobile-client__secondname"
      >
        Петрович
      </div>
      <div
        className="mobile-client__balance"
      >
        -2200
      </div>
      <div
        className="mobile-client__active"
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      >
        blocked
      </div>
      <input
        onClick={[Function]}
        type="button"
        value="Edit"
      />
    </div>
    <div
      className="mobile-client mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Довгель
      </div>
      <div
        className="mobile-client__name"
      >
        Анастасия
      </div>
      <div
        className="mobile-client__secondname"
      >
        Генриховна
      </div>
      <div
        className="mobile-client__balance"
      >
        0
      </div>
      <div
        className="mobile-client__active"
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      >
        blocked
      </div>
      <input
        onClick={[Function]}
        type="button"
        value="Edit"
      />
    </div>
  </div>
  <input
    onClick={[Function]}
    type="button"
    value="Add client"
  />
</div>
`;

exports[`работа MobileCompany 4`] = `
<div
  className="mobile-company"
>
  <h1
    className="mobile-company__name"
  >
    Компания «
    Velcom
    »
  </h1>
  <hr />
  <input
    onClick={[Function]}
    type="button"
    value="All"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Active"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Deactive"
  />
  <hr />
  <div
    className="mobile-company__client"
  >
    <div
      className="mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Фамилия
      </div>
      <div
        className="mobile-client__name"
      >
        Имя
      </div>
      <div
        className="mobile-client__fathername"
      >
        Отчество
      </div>
      <div
        className="mobile-client__balance"
      >
        Баланс
      </div>
      <div
        className="mobile-client__status"
      >
        Статус
      </div>
      <div
        className="mobile-client__edit"
      >
        Редактировать
      </div>
    </div>
    <div
      className="mobile-client mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Иванов
      </div>
      <div
        className="mobile-client__name"
      >
        Иван
      </div>
      <div
        className="mobile-client__secondname"
      >
        Иванович
      </div>
      <div
        className="mobile-client__balance"
      >
        200
      </div>
      <div
        className="mobile-client__active"
        style={
          Object {
            "backgroundColor": "green",
          }
        }
      >
        active
      </div>
      <input
        onClick={[Function]}
        type="button"
        value="Edit"
      />
    </div>
    <div
      className="mobile-client mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Сидоров
      </div>
      <div
        className="mobile-client__name"
      >
        Сидор
      </div>
      <div
        className="mobile-client__secondname"
      >
        Сидорович
      </div>
      <div
        className="mobile-client__balance"
      >
        2700
      </div>
      <div
        className="mobile-client__active"
        style={
          Object {
            "backgroundColor": "green",
          }
        }
      >
        active
      </div>
      <input
        onClick={[Function]}
        type="button"
        value="Edit"
      />
    </div>
    <div
      className="mobile-client mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Петров
      </div>
      <div
        className="mobile-client__name"
      >
        Петр
      </div>
      <div
        className="mobile-client__secondname"
      >
        Петрович
      </div>
      <div
        className="mobile-client__balance"
      >
        -2200
      </div>
      <div
        className="mobile-client__active"
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      >
        blocked
      </div>
      <input
        onClick={[Function]}
        type="button"
        value="Edit"
      />
    </div>
    <div
      className="mobile-client mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Прупас
      </div>
      <div
        className="mobile-client__name"
      >
        Матвей
      </div>
      <div
        className="mobile-client__secondname"
      >
        Викторович
      </div>
      <div
        className="mobile-client__balance"
      >
        100500
      </div>
      <div
        className="mobile-client__active"
        style={
          Object {
            "backgroundColor": "green",
          }
        }
      >
        active
      </div>
      <input
        onClick={[Function]}
        type="button"
        value="Edit"
      />
    </div>
    <div
      className="mobile-client mobile-client__row"
    >
      <div
        className="mobile-client__lastname"
      >
        Довгель
      </div>
      <div
        className="mobile-client__name"
      >
        Анастасия
      </div>
      <div
        className="mobile-client__secondname"
      >
        Генриховна
      </div>
      <div
        className="mobile-client__balance"
      >
        0
      </div>
      <div
        className="mobile-client__active"
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      >
        blocked
      </div>
      <input
        onClick={[Function]}
        type="button"
        value="Edit"
      />
    </div>
  </div>
  <input
    onClick={[Function]}
    type="button"
    value="Add client"
  />
</div>
`;
